pkg/config/config.go

```diff
@@ -406,6 +406,9 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("kubernetes_map_services_on_ip", false) // temporary opt-out of the new mapping logic
 	config.BindEnvAndSetDefault("kubernetes_apiserver_use_protobuf", false)
 
+	config.BindEnvAndSetDefault("prometheus_scrape.enabled", false)
```

Comment:
Mind adding a comment here?

Reply:
I think we don't need this config actually, it should be a helm/operator config option to enable the new config provider via `DD_EXTRA_CONFIG_PROVIDERS` behind the scenes. WDYT?
I'm removing it for now, I'll add a comment on `prometheus_scrape.checks` though